### ISSUE TITLE
DateFormatConstant check should not warn for fields annotated by GuardedBy

### ISSUE THREAD

**Issue by msridhar** [google/error-prone#906]
Looks like the new DateFormatConstant check does not account for other thread safety annotations.

### What version of Error Prone are you using?

2.2.0

### Does this issue reproduce with the latest release?

Yes

### What did you do?

Compiled code like this:
```java
@ThreadSafe
public class A {

  @GuardedBy("A.class")
  private static final DateFormat DATE_FORMAT =
      new SimpleDateFormat("yyyyMMdd-HHmmss", Locale.US);
}
```

### What did you expect to see?

No warning, since there are explicit `@ThreadSafe` and `@GuardedBy` annotations.

### What did you see instead?

A warning from the DateFormatConstant check.
